### DISCUSSION TITLE
Questions about index continuity where setBindGroup(index, ...)

### DISCUSSION THREAD

**Discussion by ghost** [gpuweb/gpuweb#3043] [Ideas/Feedback]
Now wgsl as follow: 

``` wgsl
   @group(3) @binding(1) var mySampler: sampler;
   @group(3) @binding(2) var myTexture: texture_2d<f32>;
```

I createPipelineLayout, the 0, 1, 2 must create a new BindGroupLayout with empty entry instead of null; otherwise the createpiepeLineLayout will failed;

I get a bindgroup with createBindGroup, and then I call rp.setBindGroup(3, bindGroup); without 0, 1, 2 when I run wgpu, I get a error : not set BindGroup 0, not set BindGroup 1, not set BindGroup 2

I can't find the reference about the WebGPU Spec, can't you help me? 

I do that because the shader must be flexible when handle much feature by macro "ifdef ... else if ... else ", which I impl it by regex in js

**Comment by Kangz** [comment]
Right now in the spec you need to bind empty bind groups to slot 0, 1 and 2 to be able to use bindgroup 3.

However there has been proposals to relax this in #2043

**Comment by ghost** [comment]
Thanks vevy much for answer me.
